lib/checktype.cpp

```diff
@@ -425,7 +426,7 @@ void CheckType::checkFloatToIntegerOverflow(const Token *tok, const ValueType *v
             continue;
         if (!mSettings->isEnabled(&f, false))
             continue;
-        if (f.floatValue > ~0ULL)
+        if (f.floatValue >= std::pow(2., std::numeric_limits<unsigned long long>::digits))
             floatToIntegerOverflowError(tok, f);
         else if ((-f.floatValue) > (1ULL<<62))
```

Comment:
Shouldn't this be a 63 instead of 62? A `long long` is able to represent any number between -(2^63 - 1) and (2^63 - 1), including the bounds, or any number between -(2^63) and (2^63 - 1) as of C++20.

Reply:
hmm yes I think so too.